core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -1679,6 +1679,15 @@ package object config {
       .checkValues(Set("ADLER32", "CRC32", "CRC32C"))
       .createWithDefault("ADLER32")
 
+  private[spark] val SCHEDULER_CHECKSUM_MISMATCH_FULL_RETRY_ENABLED =
+    ConfigBuilder("spark.scheduler.checksumMismatchFullRetry.enabled")
+      .doc("Whether to retry all tasks of a consumer stage when we detect checksum mismatches " +
+        "with its producer stages. The checksum computation is controlled by another config " +
+        "called SHUFFLE_ORDER_INDEPENDENT_CHECKSUM_ENABLED.")
```

Comment:
Does it make sense to use `SHUFFLE_ORDER_INDEPENDENT_CHECKSUM_ENABLED` without `SCHEDULER_CHECKSUM_MISMATCH_FULL_RETRY_ENABLED `? or vice versa?

What about getting removing the `SHUFFLE_ORDER_INDEPENDENT_CHECKSUM_ENABLED` (as the version is where it is introduced is also 4.1.0 we can do that) and  computing the checksum when `SCHEDULER_CHECKSUM_MISMATCH_FULL_RETRY_ENABLED ` is true? So having only one config for the feature?